nextcord/application_command.py

```diff
@@ -3276,3 +3293,177 @@ def unpack_annotation(
         raise ValueError(f"Given Annotation {given_annotation} has an unhandled origin: {origin}")
 
     return type_ret, literal_ret
+
+
+class Range(int):
+    """An annotation helper for defining slash command `min_value` and `max_value` parameters.
+
+    .. versionadded:: 2.2
+
+    .. container:: operations
+
+        .. describe:: Range[x, y]
+
+            Creates a range from ``x`` to ``y``.
+
+        .. describe:: Range[x] | Range[..., x]
+
+            Create a range up to ``x``.
+
+        .. describe:: Range[x, ...]
+
+            Create a range from ``x``.
+    """
+
+    min: ClassVar[Optional[Union[int, float]]]
+    max: ClassVar[Optional[Union[int, float]]]
+
+    @overload
+    def __class_getitem__(
+        cls,
+        value: Union[
+            int,
+            Tuple[int, int],
+            Tuple[int, EllipsisType],
+            Tuple[EllipsisType, int],
+        ],
+    ) -> Type[int]:
+        ...
+
+    @overload
+    def __class_getitem__(
+        cls,
+        value: Union[
+            float,
+            Tuple[float, float],
+            Tuple[float, EllipsisType],
+            Tuple[EllipsisType, float],
+        ],
+    ) -> Type[float]:
+        ...
+
+    def __class_getitem__(
+        cls,
+        value: Union[
+            int,
+            float,
+            Tuple[int, int],
+            Tuple[float, float],
+            Tuple[Union[int, float], EllipsisType],
+            Tuple[EllipsisType, Union[int, float]],
+        ],
+    ) -> Type[Union[int, float]]:
+        class Inner(Range, OptionConverter):
+            def __init__(self):
+                super().__init__(option_type=type(self.min or self.max))
+
+            def modify(self, option: SlashCommandOption):
+                if self.min:
+                    if option.min_value is None:
+                        option.min_value = self.min
+                if self.max:
+                    if option.max_value is None:
+                        option.max_value = self.max
+
+        if isinstance(value, tuple):
+            if len(value) != 2:
+                raise ValueError("Range can only take 1-2 arguments")
+
+            min = value[0]
+            max = value[1]
+        else:
+            min = None
+            max = value
+
+        if min is None or isinstance(min, EllipsisType):
+            Inner.min = None
+        elif isinstance(min, (int, float)):
+            Inner.min = min
+        else:
+            raise TypeError("Range min must be int or float.")
+
+        if isinstance(max, EllipsisType):
+            Inner.max = None
+        elif isinstance(max, (int, float)):
+            Inner.max = max
+        else:
+            raise TypeError("Range max must be int or float.")
+
+        if Inner.min is None and Inner.max is None:
+            raise TypeError("At least one of min or max must be set.")
+
+        return Inner
+
+
+class String(int):
```

Comment:
It isn't THAT important, but why does it subclass `int`?

Reply:
I suppose it doesnt need to anymore as `__class_getitem__` returns that, legacy code

Reply:
Would it help to subclass `str` for typing purposes or not really?

Reply:
`__class_getitem__` pretends to return `str/int/float` so the types are recognised like that. This was done in the early stage of this PR when this used a metaclass